### ISSUE TITLE
DSR support for containerd

### ISSUE THREAD

**Issue by andrewrynhard** [cloudnativelabs/kube-router#843]
I'd like to use DSR with containerd instead of Docker.

**Comment by aauren** [comment]
Maybe, I'm missing something here, but I don't believe that there is anything specifically tied to docker about our current DSR implementation. Can you explain more about what specifically doesn't work about DSR with containerd that does work with Docker?

**Comment by andrewrynhard** [comment]
@aauren The docs say this:

```
Also host path /var/run/docker.sock must be made a volumemount to kube-router.
```

**Comment by aauren** [comment]
Thanks for the reply. I had missed that in the documentation. Looking through the code, it appears that the kube-router code is pretty strongly tied to a docker runtime for handling DSR so it would probably be a bit of work to support other runtimes directly like containerd or cri-o.

Being realistic, I don't know that this is something that we'll have time to get to in the next couple of phases. Is this something that you would be interested in submitting and implementation for?

**Comment by andrewrynhard** [comment]
@aauren As much as I would :heart to, unfortunately no. I have quite a few open source projects I am juggling, and we have since learned to live without DSR. If I get a chance, I might take a shot at it, but I can't say I can commit to this right now. Maybe in the future.

**Comment by aauren** [comment]
@andrewrynhard Thanks for getting back about this. As much as it would be awesome to be able to support multiple container runtimes, we're currently focused on enhancing other areas of kube-router so I'm going to close this for now.

If someone happens across this in the future and wants to contribute the functionality just respond here, totally open to re-opening it and discussing if someone has the time to contribute this functionality.

**Comment by murali-reddy** [comment]
Current implementation support docker only. There is low level plumbing required to support DSR

https://github.com/cloudnativelabs/kube-router/blob/v1.0.0/pkg/controllers/proxy/network_services_controller.go#L869-L874

which requires to use container runtime API. If there is any library that can abstract the container runtimes it would be nice to do it once for all.

**Comment by sfxworks** [comment]
It should be noted that support for docker on k8s is going away https://github.com/kubernetes/kubernetes/pull/94624
All communication will have be CRI compatible so support for containerd crio and others will be wanted

**Comment by aauren** [comment]
Re-opening now that @ep4eg has put in a PR

**Comment by sfxworks** [comment]
Noted the merge. I'll test crio and containerd shortly.

**Comment by yydzhou** [comment]
> Noted the merge. I'll test crio and containerd shortly.

@sfxworks any update on this?

**Comment by sfxworks** [comment]
@yydzhou Hi there. Very sorry as I got caught up in a project at work. I can probably get a test on this before the weekend.

**Comment by aauren** [comment]
Closed via #1027
